pytest_splunk_addon/standard_lib/index_tests/test_templates.py

```diff
@@ -58,36 +50,89 @@ def test_index_time_extractions(
+            + ")"
+        )
+
+        if splunk_indextime_time.get("identifier"):
+            extra_filter = splunk_indextime_time.get("identifier")
+        else:
+            extra_filter = "host=" + splunk_indextime_time.get("host", "*")
 
-                msg = "looking for {}={} Found {}={}".format(
-                    key,
-                    splunk_indextime_fields["tokenized_event"].key_fields[key],
-                    key,
-                    result[key],
-                )
+        if splunk_indextime_time["tokenized_event"].time_values:
+            extra_filter += " | eval e_time=_time"
 
-                assert (
-                    result[key]
-                    in splunk_indextime_fields["tokenized_event"].key_fields[key]
-                ), msg
+        query = "sourcetype={} {} | table {}".format(
+            splunk_indextime_time.get("sourcetype"),
+            extra_filter,
+            "e_time",
+        )
 
-        # Testing line breaker
+        search = "search {} {}".format(index_list, query)
 
-        expected_events_count = int(splunk_indextime_fields[
+        results = splunk_search_util.getFieldValuesList(search)
+        results = list(results)
+        result_fields = {key: [item[key] for item in results] for key in results[0].keys()}
+
+        key_time = splunk_indextime_time["tokenized_event"].time_values
+        for index, event_time in enumerate(result_fields["e_time"]):
+            if splunk_indextime_time["tokenized_event"].metadata.get("timestamp_type") in ('event'):
+                time_diff = int(key_time[index]) - int(event_time)
+            else:
+                time_diff = int(key_time[0]) - int(event_time)
+            assert (
+                time_diff < MAX_TIME_DIFFERENCE
+            )
+
+    # Testing line breaker
+    @pytest.mark.splunk_indextime
+    def test_indextime_line_breaker(
+        self,
+        splunk_search_util,
+        splunk_indextime_line_breaker,
+        record_property,
+        caplog,
+    ):
+        expected_events_count = int(splunk_indextime_line_breaker[
             "tokenized_event"
         ].metadata.get("expected_event_count", 1))
 
-        event_count_query = "search sourcetype={} source={} host={}".format(
-            splunk_indextime_fields.get("sourcetype"),
-            splunk_indextime_fields.get("source"),
-            splunk_indextime_fields["tokenized_event"].sample_name + "*",
+        event_count_query = "search sourcetype={} (host=host_{}* OR host={}*)".format(
+            splunk_indextime_line_breaker.get("sourcetype"),
+            splunk_indextime_line_breaker.get('host'),
+            splunk_indextime_line_breaker.get('host'),
         )
-        record_property("count_query", event_count_query)
+
         assert splunk_search_util.checkQueryCount(
             event_count_query, expected_events_count, retries=0, interval=0
         ), ("We should get exactly " + str(expected_events_count) + " result")
```

Comment:
In the failure message, we should add `Found {}` as well.

Reply:
`checkQueryCount` returns boolean other ways to achieve is to fetch list of all the results compare length and show results is assertion fails, that maybe a lot of events in some cases. Anyhow we already provide the query the user can check the results in the splunk.